subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/ComponentState.java

```diff
@@ -301,6 +297,15 @@ public boolean isRejected() {
         return rejected;
     }
 
+    @Override
+    public boolean isRoot() {
```

Comment:
:o: This operation could be pretty expensive. As soon as we have 2 selectors, this should be called, and involves the creation of an iterator. We should check if it shows up in performance graphs.

Reply:
This method will be removed in the next PR, since this merge logic is no longer used.